### PR TITLE
Recentacts

### PR DESCRIPTION
**Pull request by ethn**
This is a start to a new act-based handling of recent changes.  

Other things we *ought to have*:
1. better filtering of cards we don't care about
2. grouping multiple acts on the same card(?)
3. interface touchups

### CODE REVIEW

File: card/mod/01_core/spec/set/all/content_spec.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@
       @card.save_content_draft('booboo')
       @card.reload
       expect(@card.drafts.length).to eq 1
-      expect(@card.drafts[0].new_values[:content]).to eq 'booboo'
+      expect(@card.drafts[0].value(:db_content)).to eq 'booboo'
```
Comment (ethn) [inline card/mod/01_core/spec/set/all/content_spec.rb]: also supports `value(:content)`

File: card/mod/01_history/lib/card/action.rb [hunk 2]
```diff
@@ -61,47 +69,46 @@ def delete_old
...
+    def changes
+      @changes ||=
+        begin
+          hash = {}
+          card_changes.each do |change|
+            hash[change.field.to_sym] = change
+          end
+          hash
+        end
     end
```
Comment (ethn) [inline card/mod/01_history/lib/card/action.rb]: the core of the new approach is that the caching is more on-demand, and further behind the API

File: card/mod/01_history/lib/card/action.rb [hunk 3]
```diff
@@ -112,25 +119,35 @@ def field_index field
...
 
     def new_content?
-      new_value_for :db_content
+      value :db_content
     end
 
     def new_name?
-      new_value_for :name
+      value :name
     end
```
Comment (ethn) [inline card/mod/01_history/lib/card/action.rb]: I've been working out this "new" vocab.  I understand where it came from, but it's an easier translation to the database if we refer to the "new" value as just the "value" of the change.  (it's the value field of the change row, after all")

File: card/mod/01_history/lib/card/change.rb [hunk 4]
```diff
@@ -31,7 +31,7 @@ def field= value
     end
 
     def field
-      TRACKED_FIELDS[read_attribute(:field)]
+      TRACKED_FIELDS[ attribute(:field)]
```
Comment (ethn) [inline card/mod/01_history/lib/card/change.rb]: this was not intentional, but I guess it still works??

**Comment by ethn** [comment]
I added code to address the needs for handling permissions and bad data that I mentioned before.  I'd still rather ensure we're not getting bad data, but for now those checks are in place.

File: card/mod/01_history/lib/card/action.rb (hunk 2 above)
Comment (xithan) [inline card/mod/01_history/lib/card/action.rb]: just a style remark: you can save five lines if you use `each_with_object`

**Pull request merged**